crates/bevy_asset/src/loader.rs

```diff
@@ -541,6 +541,62 @@ impl<'a> LoadContext<'a> {
         self.loader_dependencies.insert(path, hash);
         Ok(loaded_asset)
     }
+
+    /// Loads the asset at the given `path` directly from the provided `reader`. This is an async function that will wait until the asset is fully loaded before
+    /// returning. Use this if you need the _value_ of another asset in order to load the current asset, and that value comes from your [`Reader`].
+    /// For example, if you are deriving a new asset from the referenced asset, or you are building a collection of assets. This will add the `path` as a
+    /// "load dependency".
+    ///
+    /// If the current loader is used in a [`Process`] "asset preprocessor", such as a [`LoadAndSave`] preprocessor,
+    /// changing a "load dependency" will result in re-processing of the asset.
+    ///
+    /// [`Process`]: crate::processor::Process
+    /// [`LoadAndSave`]: crate::processor::LoadAndSave
+    pub async fn load_direct_with_reader<'b>(
```

Comment:
What about a `load_with_reader`, and `load_with_reader_and_settings`? Looking at the `load` and `load_with_settings`, I think it should be possible to build on top of `load_direct_with_reader` and IMO would be disproportionately useful.

Reply:
I've attempted to create these methods on `LoadContext`, but that required creating matching methods on `AssetServer` to have consistent behaviour. This still doesn't make any breaking changes, but there's now substantially more changes. Additionally, I believe there's a new foot-gun where if a user calls `load_with_reader` but `should_load_dependencies` is `false`. I don't think it makes sense to pass a `Reader` to a loader, only for it to be dropped, but maybe there are cases where you'd want to do that? I've added an `error!` message for now just to flag it for review from you guys.